app/consolidated-view-poc/react/app/useQuery.js

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react'
+import { useToken } from './AuthProvider.js'
+
+export function useLazyQuery(query, { headers, preloaded = null }) {
+  const [loading, setLoading] = useState(false)
+  const [data, setData] = useState(preloaded)
+  const [error, setError] = useState(null)
+  const [preloadedReturned, setPreloadedReturned] = useState(!preloaded)
+  const { getToken } = useToken()
+
+  return [
```

Comment:
Is the use of an array a common pattern here? I'd have thought an object with named properties would be a better API 🤷